specification/azsadmin/resource-manager/containerregistry/Microsoft.ContainerRegistry.Admin/preview/2019-11-01/setup.json

```diff
@@ -0,0 +1,223 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "2019-11-01",
+    "title": "ContainerRegistryManagementClient",
+    "description": "Admin Container Registry Management Client."
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "paths": {
+    "/subscriptions/{subscriptionId}/providers/Microsoft.ContainerRegistry.Admin/locations/{location}/setup": {
+      "post": {
+        "x-ms-examples": {
+          "Invokes container registry certificate uploading and ACR service deployment.": {
+            "$ref": "./examples/Setup/Start.json"
+          }
+        },
+        "tags": [
+          "ContainerRegistrySetupStart"
+        ],
+        "description": "Invokes container registry certificate uploading and service deployment.",
+        "operationId": "ContainerRegistrySetup_Start",
+        "parameters": [
+          {
+            "$ref": "containerregistry.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "containerregistry.json#/parameters/LocationParameter"
+          },
+          {
+            "$ref": "containerregistry.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "startSetupRequest",
+            "in": "body",
+            "description": "The object containing information for the configuration request.",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/ContainerRegistrySetupProperty"
+            }
+          }
+        ],
+        "responses": {
+          "201": {
```

Comment:
Yes, it is fine for this case

Reply:
Mark this API is likely going to be deprecated in future and got added as a stop gap . It returns Synch and there is a separate get Status API. It could have been modelled as a arm long running operation with Location header. We are not stressing too much as we see this api getting deprecated